src/ai/backend/manager/models/scaling_group.py

```diff
@@ -377,7 +383,9 @@ async def load_by_group(
             scaling_groups.c.name == sgroups_for_groups.c.scaling_group,
         )
         query = (
-            sa.select([scaling_groups]).select_from(j).where(sgroups_for_groups.c.group == group)
+            sa.select([scaling_groups])
+            .select_from(j)
+            .where(sgroups_for_groups.c.group == group and scaling_groups.c.is_public.is_(True))
```

Comment:
This code can raise TypeError("Boolean value of this clause is not defined")
```suggestion
            .where(sgroups_for_groups.c.group & group and scaling_groups.c.is_public.is_(True))
```
ref: https://stackoverflow.com/questions/42681231/sqlalchemy-unexpected-results-when-using-and-and-or